device_e2e/aio/test_send_message_stress.py

```diff
@@ -0,0 +1,597 @@
+ALL_AT_ONCE_MESSAGE_COUNT = 3000
+ALL_AT_ONCE_TOTAL_ELAPSED_TIME_FAILURE_TRIGGER = 5 * 60
+
+# Settings that apply to flaky network telemetry test
+SEND_TELEMETRY_FLAKY_NETWORK_TEST_DURATION = 5 * 60
+SEND_TELEMETRY_FLAKY_NETWORK_MESSAGES_PER_SECOND = 20
+SEND_TELEMETRY_FLAKY_NETWORK_KEEPALIVE_INTERVAL = 10
+SEND_TELEMETRY_FLAKY_NETWORK_CONNECTED_INTERVAL = 15
+SEND_TELEMETRY_FLAKY_NETWORK_DISCONNECTED_INTERVAL = 15
+
+# Settings that apply to fault injection telemetry test
+SEND_TELEMETRY_FAULT_INJECTION_TEST_DURATION = 5 * 60
+SEND_TELEMETRY_FAULT_INJECTION_MESSAGES_PER_SECOND = 20
+SEND_TELEMETRY_FAULT_INJECTION_FAULT_INTERVAL = 10
+
+
+@pytest.mark.stress
+@pytest.mark.describe("Client Stress")
+class TestSendMessageStress(object):
+    async def send_and_verify_single_telemetry_message(
+        self, client, service_helper, stress_measurements
+    ):
+        """
+        Send a single message and verify that it gets received by EventHub
+        """
+
+        random_message = get_random_message(TELEMETRY_PAYLOAD_SIZE)
+
+        # We keep track of outstanding messages by message_id.  This is useful when reading
+        # logs on failure because it lets us know _which_ messages didn't finish.
+        self.outstanding_message_ids.add(random_message.message_id)
+
+        sent = False
+        attempt = 1
+        start_time = time.time()
+
+        with stress_measurements.lock:
+            stress_measurements.queued_telemetry_messages += 1
+            stress_measurements.peak_queued_telemetry_messages = max(
+                stress_measurements.peak_queued_telemetry_messages,
+                stress_measurements.queued_telemetry_messages,
+            )
+
+        # "Poor-man's" retry policy.  retry 5 times with linear backoff
+        while not sent:
+            try:
+                # If we're not connected, we need to connect.  This could be done better.
+                # If we're sending 10 messages per second, this will give us ten calls to
+                # client.connect when we really only need one.
+                if not client.connected:
+                    logger.info("Connecting for  message {}".format(random_message.message_id))
+                    await client.connect()
+
+                logger.info("Sending message {}".format(random_message.message_id))
+                await client.send_message(random_message)
+                logger.info("Done sending message {}".format(random_message.message_id))
+
+                sent = True
+
+            except (
+                ConnectionFailedError,
+                ConnectionDroppedError,
+                OperationCancelled,
+                NoConnectionError,
+            ) as e:
+                # These are the errors that client.connect or client.send_message can raise
+                # if we're not connected.  Can we limit this list?  Surely, we don't expect
+                # ConnectoinDroppedError if auto_connect==True.  Do we want to fail if we do?
```

Comment:
If `ConnectionDroppedError` is not supposed to happen when `auto_connect` is `True`, we should fail IMO